go/pkg/client/cas.go

```diff
@@ -900,43 +892,79 @@ func (wt *writerTracker) Close() error {
 	return nil
 }
 
-func (c *Client) readBlobStreamed(ctx context.Context, hash string, sizeBytes, offset, limit int64, w io.Writer) (int64, error) {
-	if sizeBytes == 0 {
-		// Do not download empty blobs.
-		return 0, nil
-	}
-	wt := newWriteTracker(w)
-	name, wc, done, err := c.maybeCompressReadBlob(hash, sizeBytes, wt)
-	if err != nil {
-		return 0, err
-	}
-	n, err := c.readStreamed(ctx, name, offset, limit, wc)
+type resettableWriter interface {
+	io.Writer
+	Reset() error
+}
+
+type fileWriter struct {
+	f *os.File
+}
+
+func (fw *fileWriter) Write(p []byte) (int, error) {
+	return fw.f.Write(p)
+}
+
+func (fw *fileWriter) Reset() error {
+	err := fw.f.Truncate(0)
 	if err != nil {
-		return n, err
-	}
-	if err = wc.Close(); err != nil {
-		return n, err
-	}
-	if err := <-wt.ready; err != nil {
-		return n, err
-	}
-	close(wt.ready)
-	sz := sizeBytes - offset
```

Comment:
Did you drop this becasue it's arguably redundant with hash checking?
I think we should keep it anyway - and actually move it before the hash check. I just heard a thought that we can maybe have issues due to truncated reads which lead to hash mismatches.

Reply:
Done.